### PR TITLE
Use decorated component for handling messages

### LINKED ISSUE #65: Heads are not displayed when using HEURISTIC_ONLY on servers using disguised messages

### PR DESCRIPTION
**Pull request by Suiranoil**
Fixes #65

### CODE REVIEW

**Comment by Fourmisain** [comment]
Never ran into #65, how does this fix this / how does this happen in the first place?

From what I can see, `decorate` is the part that is responsible to put together e.g. sender + content into the "translated" format `<sender> content`, so that looks very important.

However, what confuses me is that the component `chatheads$handleAddedPlayerMessage` uses stems from `showMessageToPlayer`, which stems from `handlePlayerChatMessage` and is defined as `bound.decorate(playerChatMessage2.decoratedContent())`.
Meaning it already is decorated...?!

So this part shouldn't do anything, right? (And should be removed.)

Going by this, the original issue must be caused by "disguised" chat messages (aka non-player messages), which I remember being extremely weird, since they are very similar to system messages without technically being ones.
Those messages do in fact miss their decorations, so that part seems correct, but I still don't understand why _disguised_ chat messages would even have a name parameter.
I guess everything is possible in mod/plugin land.

**tl;dr**
Can you test if it still works by removing the first `bound.decorate()`?

**Comment by Fourmisain** [comment]
I've refactored the whole heuristic logic to always use the sender decoration in place of the whole message if possible (fixing the issue with undecorated disguised messages too).

I can't really test it by myself since I don't know the server plugin being used, so if you want you can test what I pushed onto the [test branch](https://github.com/dzwdz/chat_heads/tree/test).

**Comment by Suiranoil** [comment]
It seems to be resolved with the changed you made
